conan/tools/env/environment.py

```diff
@@ -486,11 +495,43 @@ def save_sh(self, file_location, generate_deactivate=True):
         content = "\n".join(result)
         save(file_location, content)
 
+    def save_fish(self, file_location, generate_deactivate=True):
```

Comment:
Note sure how to add tests for this. Locally I have one that expects to have fish available, but that's not true ion most cases. I know there's a config for these kinds of stuff, but I'm guessing if adding a new entry there would be worth it?

Reply:
It is difficult to have fully functional tests for this, we would need to install fish in the CI servers. I guess we could start with some basic integration tests, and checking the generated file important parts could be enough.